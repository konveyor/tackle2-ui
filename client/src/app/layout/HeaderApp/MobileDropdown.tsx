import React, { useState } from "react";
import { DropdownItem } from "@patternfly/react-core";
import HelpIcon from "@patternfly/react-icons/dist/esm/icons/help-icon";
import { KebabDropdown } from "@app/components/KebabDropdown";
import { AppAboutModal } from "@app/layout/AppAboutModal";

export const MobileDropdown: React.FC = () => {
  const [isAboutModalOpen, setAboutModalOpen] = useState(false);

  const toggleAboutModal = () => {
    setAboutModalOpen((current) => !current);
  };

  return (
    <>
      <KebabDropdown
        ariaLabel="mobile dropdown"
        dropdownItems={[
          <DropdownItem key="about" onClick={toggleAboutModal}>
            <HelpIcon />
            &nbsp;About
          </DropdownItem>,
        ]}
      />
      <AppAboutModal isOpen={isAboutModalOpen} onClose={toggleAboutModal} />
    </>
  );
};
