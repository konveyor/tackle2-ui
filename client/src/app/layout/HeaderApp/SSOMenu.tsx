import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { useKeycloak } from "@react-keycloak/web";
import {
  Dropdown,
  MenuToggle,
  DropdownGroup,
  DropdownItem,
  DropdownList,
  ToolbarItem,
} from "@patternfly/react-core";
import { isAuthRequired, LocalStorageKey } from "@app/Constants";
import { useHistory } from "react-router-dom";

export const SSOMenu: React.FC = () => {
  const { t } = useTranslation();

  const { keycloak } = (isAuthRequired && useKeycloak()) || {};
  const history = useHistory();
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);

  const onDropdownSelect = () => {
    setIsDropdownOpen((current) => !current);
  };

  const onDropdownToggle = (isOpen: boolean) => {
    setIsDropdownOpen(isOpen);
  };
  return (
    <>
      {keycloak && (
        <ToolbarItem
          visibility={{
            default: "hidden",
            md: "visible",
          }} /** this user dropdown is hidden on mobile sizes */
        >
          <Dropdown
            isPlain
            onSelect={onDropdownSelect}
            isOpen={isDropdownOpen}
            toggle={(toggleRef) => (
              <MenuToggle
                ref={toggleRef}
                id="sso-actions-toggle"
                onClick={() => onDropdownToggle(!isDropdownOpen)}
              >
                {(keycloak?.idTokenParsed as any)["preferred_username"]}
              </MenuToggle>
            )}
          >
            <DropdownGroup key="sso">
              <DropdownList>
                <DropdownItem
                  id="manage-account"
                  key="sso_user_management"
                  component="button"
                  onClick={() => keycloak.accountManagement()}
                >
                  {t("actions.manageAccount")}
                </DropdownItem>
                <DropdownItem
                  id="logout"
                  key="sso_logout"
                  onClick={() => {
                    // Clears selected persona from storage without updating it in React state so we don't re-render the persona selector while logging out.
                    // We have to clear it before logout because the redirect can happen before the logout promise resolves.
                    window.localStorage.removeItem(
                      LocalStorageKey.selectedPersona
                    );
                    {
                      keycloak
                        .logout()
                        .then((res) => {
                          history.push("/");
                        })
                        .catch((err) => {
                          history.push("/");
                        });
                    }
                  }}
                >
                  {t("actions.logout")}
                </DropdownItem>
              </DropdownList>
            </DropdownGroup>
          </Dropdown>
        </ToolbarItem>
      )}
    </>
  );
};
