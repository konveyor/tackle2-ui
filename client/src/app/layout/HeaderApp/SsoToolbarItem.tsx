import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { useKeycloak } from "@react-keycloak/web";
import {
  Dropdown,
  MenuToggle,
  DropdownGroup,
  DropdownItem,
  DropdownList,
  ToolbarItem,
} from "@patternfly/react-core";
import { isAuthRequired, LocalStorageKey } from "@app/Constants";
import { useHistory } from "react-router-dom";

export const SsoToolbarItem: React.FC = () => {
  return isAuthRequired ? <AuthEnabledSsoToolbarItem /> : <></>;
};

export const AuthEnabledSsoToolbarItem: React.FC = () => {
  const { t } = useTranslation();

  const { keycloak } = useKeycloak();
  const history = useHistory();
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);

  const onDropdownSelect = () => {
    setIsDropdownOpen((current) => !current);
  };

  const onDropdownToggle = (isOpen: boolean) => {
    setIsDropdownOpen(isOpen);
  };

  return (
    <ToolbarItem>
      <Dropdown
        onSelect={onDropdownSelect}
        isOpen={isDropdownOpen}
        onOpenChange={onDropdownToggle}
        toggle={(toggleRef) => (
          <MenuToggle
            isFullHeight
            ref={toggleRef}
            id="sso-actions-toggle"
            onClick={() => onDropdownToggle(!isDropdownOpen)}
          >
            {keycloak?.idTokenParsed?.["preferred_username"] ??
              "DefaultUsername"}
          </MenuToggle>
        )}
      >
        <DropdownGroup key="sso">
          <DropdownList>
            <DropdownItem
              id="manage-account"
              key="sso_user_management"
              component="button"
              onClick={() => keycloak.accountManagement()}
            >
              {t("actions.manageAccount")}
            </DropdownItem>
            <DropdownItem
              id="logout"
              key="sso_logout"
              onClick={() => {
                window.localStorage.removeItem(LocalStorageKey.selectedPersona);
                {
                  keycloak
                    .logout()
                    .then(() => {
                      history.push("/");
                    })
                    .catch((err) => {
                      console.error("Logout failed:", err);
                      history.push("/");
                    });
                }
              }}
            >
              {t("actions.logout")}
            </DropdownItem>
          </DropdownList>
        </DropdownGroup>
      </Dropdown>
    </ToolbarItem>
  );
};
