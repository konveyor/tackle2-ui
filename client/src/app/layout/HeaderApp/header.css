.logo-pointer{
    cursor: default !important
}
.pf-c-page__header-brand-link {
    all: unset !important
}
.redhat-logo-style {
    height: 30px;
    vertical-align: middle;
  }