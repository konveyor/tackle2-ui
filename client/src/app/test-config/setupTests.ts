import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
