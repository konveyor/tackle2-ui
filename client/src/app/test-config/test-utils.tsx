import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
