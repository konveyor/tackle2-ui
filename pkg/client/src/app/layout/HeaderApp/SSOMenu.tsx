import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { useKeycloak } from "@react-keycloak/web";
import {
  Dropdown,
  DropdownGroup,
  DropdownItem,
  DropdownToggle,
  PageHeaderToolsItem,
} from "@patternfly/react-core";

export const SSOMenu: React.FC = () => {
  const { t } = useTranslation();

  const { keycloak } = useKeycloak();
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);

  const onDropdownSelect = () => {
    setIsDropdownOpen((current) => !current);
  };

  const onDropdownToggle = (isOpen: boolean) => {
    setIsDropdownOpen(isOpen);
  };

  return (
    <PageHeaderToolsItem
      visibility={{
        default: "hidden",
        md: "visible",
      }} /** this user dropdown is hidden on mobile sizes */
    >
      {keycloak && (
        <Dropdown
          isPlain
          position="right"
          onSelect={onDropdownSelect}
          isOpen={isDropdownOpen}
          toggle={
            <DropdownToggle onToggle={onDropdownToggle}>
              {(keycloak.idTokenParsed as any)["preferred_username"]}
            </DropdownToggle>
          }
          dropdownItems={[
            <DropdownGroup key="sso">
              <DropdownItem
                key="sso_user_management"
                component="button"
                onClick={() => keycloak.accountManagement()}
              >
                {t("actions.manageAccount")}
              </DropdownItem>
              <DropdownItem key="sso_logout" onClick={() => keycloak.logout()}>
                {t("actions.logout")}
              </DropdownItem>
            </DropdownGroup>,
          ]}
        />
      )}
    </PageHeaderToolsItem>
  );
};
