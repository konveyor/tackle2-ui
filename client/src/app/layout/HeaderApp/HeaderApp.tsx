import React from "react";
import {
  PageHeader,
  Brand,
  PageHeaderTools,
  Avatar,
  PageHeaderToolsGroup,
  PageHeaderToolsItem,
  Button,
  ButtonVariant,
  Title,
} from "@patternfly/react-core";
import HelpIcon from "@patternfly/react-icons/dist/esm/icons/help-icon";

import { AppAboutModalState } from "../AppAboutModalState";
import { SSOMenu } from "./SSOMenu";
import { MobileDropdown } from "./MobileDropdown";

import navBrandImage from "@app/images/konveyor-logo-white-text.png";
import { APP_BRAND, BrandType, isAuthRequired } from "@app/Constants";
import logoRedHat from "@app/images/logoRedHat.svg";
import "./header.css";

export const HeaderApp: React.FC = () => {
  const toolbar = (
    <PageHeaderTools>
      <PageHeaderToolsGroup
        visibility={{
          default: "hidden",
          "2xl": "visible",
          xl: "visible",
          lg: "visible",
          md: "hidden",
          sm: "hidden",
        }}
      >
        <PageHeaderToolsItem>
          <AppAboutModalState>
            {({ toggleModal }) => {
              return (
                <Button
                  id="about-button"
                  aria-label="about button"
                  variant={ButtonVariant.plain}
                  onClick={toggleModal}
                >
                  <HelpIcon />
                </Button>
              );
            }}
          </AppAboutModalState>
        </PageHeaderToolsItem>
      </PageHeaderToolsGroup>
      <PageHeaderToolsGroup>
        <PageHeaderToolsItem
          visibility={{
            lg: "hidden",
          }} /** this kebab dropdown replaces the icon buttons and is hidden for desktop sizes */
        >
          <MobileDropdown />
        </PageHeaderToolsItem>
        <SSOMenu />
      </PageHeaderToolsGroup>
      {APP_BRAND === BrandType.MTA && (
        <PageHeaderToolsGroup>
          <PageHeaderToolsItem>
            <img src={logoRedHat} alt="Logo" className="redhat-logo-style" />
          </PageHeaderToolsItem>
        </PageHeaderToolsGroup>
      )}
    </PageHeaderTools>
  );

  const headerLogo =
    APP_BRAND === BrandType.Konveyor ? (
      <Brand src={navBrandImage} alt="brand" />
    ) : (
      <Title className="logo-pointer" headingLevel="h1" size="2xl">
        Migration Toolkit for Applications
      </Title>
    );

  return <PageHeader logo={headerLogo} headerTools={toolbar} showNavToggle />;
};
