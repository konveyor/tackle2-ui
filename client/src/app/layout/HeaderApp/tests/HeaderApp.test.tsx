import { render } from "@app/test-config/test-utils";
import React from "react";
import { HeaderApp } from "../HeaderApp";

it("Test snapshot", () => {
  const wrapper = render(<HeaderApp />);
  expect(wrapper).toMatchSnapshot();
});
