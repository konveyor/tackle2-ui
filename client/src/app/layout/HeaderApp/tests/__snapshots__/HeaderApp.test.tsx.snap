// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="pf-c-page__header"
      >
        <div
          class="pf-c-page__header-brand"
        >
          <div
            class="pf-c-page__header-brand-toggle"
          >
            <button
              aria-disabled="false"
              aria-expanded="true"
              aria-label="Global navigation"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="nav-toggle"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </button>
          </div>
          <a
            class="pf-c-page__header-brand-link"
          >
            <img
              alt="brand"
              class="pf-c-brand"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="pf-c-page__header-tools"
        >
          <div
            class="pf-c-page__header-tools-group pf-m-hidden pf-m-visible-on-2xl pf-m-visible-on-xl pf-m-visible-on-lg pf-m-hidden-on-md pf-m-hidden-on-sm"
          >
            <div
              class="pf-c-page__header-tools-item"
            >
              <button
                aria-disabled="false"
                aria-label="about-button"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                id="aboutButton"
                type="button"
              >
                <test-file-stub />
              </button>
            </div>
          </div>
          <div
            class="pf-c-page__header-tools-group"
          >
            <div
              class="pf-c-page__header-tools-item pf-m-hidden-on-lg"
            >
              <div
                class="pf-c-dropdown pf-m-align-right"
                data-ouia-component-id="OUIA-Generated-Dropdown-1"
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="Actions"
                  class="pf-c-dropdown__toggle pf-m-plain"
                  id="pf-dropdown-toggle-id-0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 192 512"
                    width="1em"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="pf-c-page__header-tools-item pf-m-hidden pf-m-visible-on-md"
            />
          </div>
        </div>
      </header>
    </div>
    <div />
    <div />
  </body>,
  "container": <div>
    <header
      class="pf-c-page__header"
    >
      <div
        class="pf-c-page__header-brand"
      >
        <div
          class="pf-c-page__header-brand-toggle"
        >
          <button
            aria-disabled="false"
            aria-expanded="true"
            aria-label="Global navigation"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="nav-toggle"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 448 512"
              width="1em"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
          </button>
        </div>
        <a
          class="pf-c-page__header-brand-link"
        >
          <img
            alt="brand"
            class="pf-c-brand"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="pf-c-page__header-tools"
      >
        <div
          class="pf-c-page__header-tools-group pf-m-hidden pf-m-visible-on-2xl pf-m-visible-on-xl pf-m-visible-on-lg pf-m-hidden-on-md pf-m-hidden-on-sm"
        >
          <div
            class="pf-c-page__header-tools-item"
          >
            <button
              aria-disabled="false"
              aria-label="about-button"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="aboutButton"
              type="button"
            >
              <test-file-stub />
            </button>
          </div>
        </div>
        <div
          class="pf-c-page__header-tools-group"
        >
          <div
            class="pf-c-page__header-tools-item pf-m-hidden-on-lg"
          >
            <div
              class="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-1"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Actions"
                class="pf-c-dropdown__toggle pf-m-plain"
                id="pf-dropdown-toggle-id-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="pf-c-page__header-tools-item pf-m-hidden pf-m-visible-on-md"
          />
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
